doc/developer/design/20230519_statement_logging.md

```diff
@@ -0,0 +1,232 @@
+
+On system restart, all entries in `mz_statement_execution_history`
+older than `statement_log_max_age` are deleted. Then "garbage
+collection" is done to delete entries in
+`mz_prepared_statement_history` and `mz_session_history` that are no
+longer referenced.
+
+The schemas of the tables are as follows. Note that deciding on a
+particular schema now doesn't preclude us from adding more fields in
+the future, if we decide we want more granular information about
+particular queries (for example, whether WMR was used).
+
+**mz_prepared_statement_history**
+
+| Field         | Type                       | Meaning                                            |
+|---------------|----------------------------|----------------------------------------------------|
+| `id`          | `uuid`                     | The unique identifier of the prepared statement.   |
+| `session_id`  | `uuid`                     | The ID of the session that prepared the statement. |
+| `name`        | `text`                     | The name of the prepared statement.                |
+| `sql`         | `text`                     | The SQL text of the statement.                     |
+| `prepared_at` | `timestamp with time zone` | The timestamp at which the statement was prepared. |
+
+**mz_statement_execution_history**
+
+| Field                   | Type                       | Meaning                                                                                                  |
+|-------------------------|----------------------------|----------------------------------------------------------------------------------------------------------|
+| `id`                    | `uuid`                     | The unique identifier of the statement execution.                                                        |
+| `prepared_statement_id` | `uuid`                     | The unique identifier of the corresponding item in `mz_prepared_statement_history`.                      |
+| `sample_rate`           | `float8`                   | The value of `statement_history_sample_rate` at the time of execution.                                   |
+| `params`                | `text list`                | The textual representations of the supplied arguments.                                                   |
+| `began_at`              | `timestamp with time zone` | The time at which the coordinator received the command and began executing the statement.                |
+| `finished_at`           | `timestamp with time zone` | The time at which the statement finished, or null if it is still running.                                |
+| `was_successful`        | `bool`                     | Whether the statement terminated successfully, or null if it is still running.                           |
+| `was_canceled`          | `bool`                     | Whether the statement was canceled, or null if it is still running.                                      |
```

Comment:
What do we do about statements that were running when envd crashed?

Reply:
We can call this something like "aborted" instead of canceled or failed. Any suggestions on a better name?

Reply:
`was_aborted` SGTM. Will envd, on restart, set `was_aborted` to `true` for any queries that were not finished?

Reply:
Yes, exactly